### DISCUSSION TITLE
Disable LRU cache

### DISCUSSION THREAD

**Discussion by aamorozov** [vercel/next.js#21535] [Help]
Hi! 
I'm looking into serving the static files/doing ISSG while using CDN's cache and would like to disable the LRU cache that is set in next-server completely. That cache object sets the data for the pages when the revalidation happens, and I would like to avoid that and instead use the file system as the source of truth.
Is it possible?

Refs:
https://github.com/vercel/next.js/blob/canary/packages/next/next-server/server/incremental-cache.ts
https://github.com/vercel/next.js/blob/canary/packages/next/next-server/server/next-server.ts#L264
https://github.com/vercel/next.js/blob/canary/packages/next/next-server/server/next-server.ts#L1758

**Comment by stefanb2** [comment]
Proposed a solution in PR #27337

**Comment by stefanb2** [comment]
Another solution was proposed in PR #27335

**Comment by balazsorban44** [comment]
What is the actual purpose of this in-memory Incremental Cache? Could someone explain?

**Reply by karlmacklin** [reply]
We now know it keeps track of what urls have resulted in 404's and redirects. When this memory runs out on larger sites, previously deleted pages magically reappear:

https://github.com/vercel/next.js/issues/37945
